app/dao/inbound_sms_dao.py

```diff
@@ -18,8 +20,10 @@ def dao_create_inbound_sms(inbound_sms):
 
 
 def dao_get_inbound_sms_for_service(service_id, limit=None, user_number=None):
+    start_date = get_london_midnight_in_utc(date.today() - timedelta(days=6))
     q = InboundSms.query.filter(
-        InboundSms.service_id == service_id
+        InboundSms.service_id == service_id,
+        func.date(InboundSms.created_at) >= start_date
```

Comment:
don't call `func.date` here (or anywhere else in this PR). The `start_date` might be, for example, `2018-05-01T23:00:00` (midnight tonight), and sms created both at 12:00 and 23:30 (today and tomorrow respectively) will both have a date of `2018-05-01T00:00:00`.

You should be able to write a test that proves this as well.

Reply:
see https://github.com/alphagov/notifications-api/pull/1858 for details of what to change